### PR TITLE
Fix the bug on javascript won't run on receipt page

### PR DESCRIPTION
**Pull request by schenedx**
ECOM-7716

@clintonb @AlasdairSwan @mjfrey Please review

With `define`, I found the javascript snippet here just won't run on the page.
Update it to `require` so it can run. I was able to verify this in the Chrome dev console on stage.

### CODE REVIEW

**Review by mjfrey — APPROVED** [review]

**Comment by clintonb** [comment]
@edx/fedx-team can we get an approval, please?

**Review by clintonb — APPROVED** [review]
I'd like to get an approval from a front-end expert.

**Comment by andy-armstrong** [comment]
If the file is defining a module to be loaded by another JavaScript file, then the code was correct as it was. However, if the file is being loaded directly, then you want to use ``require`` to make the code execute. If you load a file with ``define`` then all you've done is define the module for somebody else to invoke.

http://stackoverflow.com/questions/9507606/when-should-i-use-require-and-when-to-use-define

It looks like this file was intended to be loaded and executed directly, so in that case 👍

**Comment by clintonb** [comment]
Thanks @andy-armstrong!

**Comment by clintonb** [comment]
@schenedx I restarted the tests. I sense flakiness; although, not due to this change.

File: ecommerce/static/js/apps/order_receipt_app.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require([
+        'jquery',
```
Comment (clintonb) [inline ecommerce/static/js/apps/order_receipt_app.js]: jQuery is never used. This can go away.

**Comment by schenedx** [comment]
@clintonb That failed build was not due to flakiness. It was due to the fact that the new `order_receipt_app.js` was not included into the `build.js`
It is now included. The build should pass this time.

**Comment by clintonb** [comment]
Noted. I had not noticed you changed the code beyond the original two lines fix from earlier.

**Comment by schenedx** [comment]
@clintonb With the two line change, the jasmine unit test would fail.
Therefore I have to use a wrapper `app` file to initiate the code on `receipt_page.js`. It's the pattern we use for all other javascript apps. See https://github.com/edx/ecommerce/tree/master/ecommerce/static/js/apps

**Pull request merged**